src/cargo/util/config.rs

```diff
@@ -847,21 +847,21 @@ impl fmt::Display for ConfigKey {
 /// Internal error for serde errors.
 #[derive(Debug)]
 pub struct ConfigError {
-    message: String,
+    error: failure::Error,
```

Comment:
We have `pub use failure::Error as CargoError` somewhere. I don't know if we want to use `failure::Error` or `CargoError`, but currently we use `CargoError` in the majority of the code base.

Reply:
I kinda assumed that was there just for being compatible with the old `CargoError` which used to be its own type.  @alexcrichton was it intended to continue to maintain that abstraction?  The use of `failure::Error` is being used in about 4 other places, so it is leaking a little.

Reply:
Oh that was actually a holdover to make the "migrate to `failure`" patch smaller than it otherwise would be. Nowadays we should just switch to `failure::Error` everywhere I think, the transition to `failure` just isn't fully done yet